core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala

```diff
@@ -113,6 +113,15 @@ class NettyBlockRpcServer(
             s"when there is not sufficient space available to store the block.")
           responseContext.onFailure(exception)
         }
+
+      case getLocalDirs: GetLocalDirsForExecutors =>
+        assert(getLocalDirs.appId == appId)
+        assert(getLocalDirs.execIds.length == 1)
```

Comment:
Although we have these assertions, what happen when this `NettyBlockRpcServer` receives a wrong message, `GetLocalDirsForExecutors` with an irrelevant `execId`? It seems that we trust the sender always and reply with the received `execId`.

Reply:
This's a good point. I've changed it to reply to the sender with the error if the request fails assertion. Thus, the sender could handle the error as fetch failure.